src/plugins/data_source_management/public/components/direct_query_data_sources_components/acceleration_creation/create/utils.tsx

```diff
@@ -0,0 +1,170 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import {
+  ACCELERATION_INDEX_NAME_REGEX,
+  ACCELERATION_S3_URL_REGEX,
+} from '../../../../../framework/constants';
+import {
+  AccelerationIndexType,
+  AccelerationRefreshType,
+  CreateAccelerationForm,
+  FormErrorsType,
+  MaterializedViewQueryType,
+  SkippingIndexRowType,
+} from '../../../../../framework/types';
+
+export const pluralizeTime = (timeWindow: number) => {
+  return timeWindow > 1 ? 's' : '';
+};
+
+export const hasError = (formErrors: FormErrorsType, key?: keyof FormErrorsType) => {
+  if (!key) return Object.values(formErrors).some((e) => !!e.length);
+  return !!formErrors[key]?.length;
+};
+
+export const validateDataSource = (dataSource: string) => {
+  return dataSource.trim().length === 0 ? ['Select a valid data source'] : [];
+};
+
+export const validateDatabase = (database: string) => {
+  return database.trim().length === 0 ? ['Select a valid database'] : [];
+};
+
+export const validateDataTable = (dataTable: string) => {
+  return dataTable.trim().length === 0 ? ['Select a valid table'] : [];
+};
+
+export const validatePrimaryShardCount = (primaryShardCount: number) => {
+  return primaryShardCount < 1 ? ['Primary shards count should be greater than 0'] : [];
+};
+
+export const validateReplicaCount = (replicaCount: number) => {
+  return replicaCount < 0 ? ['Replica count should be equal or greater than 0'] : [];
+};
+
+export const validateRefreshInterval = (
+  refreshType: AccelerationRefreshType,
+  refreshWindow: number,
+  refreshInterval: string
+) => {
+  if (refreshType === 'autoInterval' && refreshInterval === 'minute' && refreshWindow < 15)
```

Comment:
nit. can refreshWindow be configured through config file?

Reply:
based on what im understanding, this should not be configured in the config file. We do not have any config related to the specific index interval. By using the frontend flow, the minimum refresh interval is set to 15mins. However, if the user would like to configure this lower than 15mins it can be created by manual creation in query workbench.